### PR TITLE
[NP-137] Payment Page (Top up)

### PR DESCRIPTION
**Pull request by Petoxic**
## **Ticket**

**JIRA ID:** [NP-137](https://parking-system.atlassian.net/browse/NP-137)

## Changes proposed in this pull request
- Add top up page with payment options

please fill all the required fields from below Checklist otherwise reviewer **will not review** this PR
## Checklist
- [x] Screenshot evidence for both IOS and Android
- [x] Total change is less than 500 lines of code

## **Screenshot Evidence**

https://github.com/ParkingSystemWithLPR/Napark-application/assets/70986870/80b0b9c9-1066-46d6-9ecb-5d9ef54a15d9

https://github.com/ParkingSystemWithLPR/Napark-application/assets/70986870/ff831a7b-5cf2-4cde-ad15-cf36a08c40ab

[NP-137]: https://parking-system.atlassian.net/browse/NP-137?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: screens/payment/TopUp.tsx [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+  const rightText = rightAmount === 0 ? "Custom" : `฿ ${rightAmount}`;
+  return (
+    <View style={styles.doubleButtonContainer}>
+      <SecondaryButton
+        title={leftText}
+        onPress={() => onButtonClicked(leftAmount)}
+        outerContainerStyle={styles.doubleButton}
+        buttonStyle={{
+          backgroundColor:
+            activeButton === leftAmount ? Colors.red[400] : Colors.gray[700],
```
Comment (TaeCV) [inline screens/payment/TopUp.tsx]: gray[800]?

File: screens/payment/TopUp.tsx [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+        }}
+      />
+      <SecondaryButton
+        title={rightText}
+        onPress={() => onButtonClicked(rightAmount)}
+        outerContainerStyle={styles.doubleButton}
+        buttonStyle={{
+          backgroundColor:
+            activeButton === rightAmount ? Colors.red[400] : Colors.gray[700],
```
Comment (TaeCV) [inline screens/payment/TopUp.tsx]: this too

File: screens/payment/TopUp.tsx [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+    padding: 8,
+    paddingVertical: Platform.OS === "android" ? 13 : 16,
+    shadowColor: Colors.black,
+    shadowOffset: {
+      width: 0,
+      height: 2,
+    },
+    shadowOpacity: 0.2,
+    shadowRadius: 2,
+    elevation: Platform.OS === "android" ? 4 : 2,
```
Comment (TaeCV) [inline screens/payment/TopUp.tsx]: No need to check android anymore since IOS doesn't use this

File: screens/payment/TopUp.tsx [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+      <DoubleButton
+        leftAmount={1000}
+        rightAmount={0}
+        onButtonClicked={onPressButton}
+        activeButton={activeButton}
+      />
+      <PrimaryButton
+        title="Next"
+        onPress={() => {
+          if (Number(amount) >= 50) {
```
Comment (TaeCV) [inline screens/payment/TopUp.tsx]: +amount

File: screens/payment/TopUp.tsx [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+    <BodyContainer>
+      <SubHeaderText text="Current Balance" />
+      <View style={styles.currentBalance}>
+        <HeaderText text={`฿ ${route.params.balance}`} />
+      </View>
+      {isManualChooseAmount && (
+        <TextInput
+          title="Please enter amount (50 baht minimum)"
+          errorText={
+            Number(amount) < 50 ? "Please top up at least 50 baht." : ""
```
Comment (TaeCV) [inline screens/payment/TopUp.tsx]: +amount instead of Number(amount)?

File: screens/payment/TopUp.tsx (hunk 1 above)
Comment (Petoxic) [reply screens/payment/TopUp.tsx]: Fixed on [7d6023c](https://github.com/ParkingSystemWithLPR/Napark-application/pull/33/commits/7d6023c81bc08f905c97584935b65d69ed7b5de8)

File: screens/payment/TopUp.tsx (hunk 2 above)
Comment (Petoxic) [reply screens/payment/TopUp.tsx]: Fixed on [7d6023c](https://github.com/ParkingSystemWithLPR/Napark-application/pull/33/commits/7d6023c81bc08f905c97584935b65d69ed7b5de8)

File: screens/payment/TopUp.tsx (hunk 3 above)
Comment (Petoxic) [reply screens/payment/TopUp.tsx]: Fixed on [7d6023c](https://github.com/ParkingSystemWithLPR/Napark-application/pull/33/commits/7d6023c81bc08f905c97584935b65d69ed7b5de8)

File: screens/payment/TopUp.tsx (hunk 4 above)
Comment (Petoxic) [reply screens/payment/TopUp.tsx]: Fixed on [7d6023c](https://github.com/ParkingSystemWithLPR/Napark-application/pull/33/commits/7d6023c81bc08f905c97584935b65d69ed7b5de8)

File: screens/payment/TopUp.tsx (hunk 5 above)
Comment (Petoxic) [reply screens/payment/TopUp.tsx]: Fixed on [7d6023c](https://github.com/ParkingSystemWithLPR/Napark-application/pull/33/commits/7d6023c81bc08f905c97584935b65d69ed7b5de8)

**Pull request merged**